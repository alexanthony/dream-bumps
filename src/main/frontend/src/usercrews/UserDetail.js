// @flow
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { Card, CardTitle, CardText } from 'material-ui/Card';
import { Table, TableBody, TableRow, TableRowColumn } from 'material-ui/Table';

import { getUser } from '../rootReducer';
import { fetchUsers } from '../admin/actions';
import { getCollegeName } from '../utils/colleges';

/**
 * CurrentUserCrew
 */
class UserDetail extends Component {
  componentDidMount() {
    this.props.fetchUsers();
  }

  render() {
    return (
      <Card>
        <CardTitle title="User Details" />
        <CardText>
          <Table selectable={false}>
            <TableBody displayRowCheckbox={false}>
              <TableRow>
                <TableRowColumn>Username</TableRowColumn>
                <TableRowColumn>{this.props.user.firstName}</TableRowColumn>
              </TableRow>
              <TableRow>
                <TableRowColumn>College</TableRowColumn>
                <TableRowColumn>
                  {getCollegeName(this.props.user.college)}
                </TableRowColumn>
              </TableRow>
            </TableBody>
          </Table>
        </CardText>
      </Card>
    );
  }
}

UserDetail.propTypes = {
  userId: PropTypes.number,
  fetchUser: PropTypes.func,
  user: PropTypes.shape({
    firstName: PropTypes.string,
    college: PropTypes.string
  })
};

export const mapStateToProps = (
  state: Object,
  ownProps: { userId: number, sex: string }
) => ({
  user: getUser(state)(ownProps.userId)
});

export { UserDetail };

export default connect(mapStateToProps, { fetchUsers })(UserDetail);
