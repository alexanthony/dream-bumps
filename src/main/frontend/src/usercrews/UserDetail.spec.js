import React from 'react';
import { shallow } from 'enzyme';
import { UserDetail } from './UserDetail';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <UserDetail userId={2} user={{ firstName: 'UserName', college: 'newc' }} />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot();
  // We don't actually mount, so these don't get called...
  // expect(fetchUsers).toHaveBeenCalled();
  // expect(fetchUserCrewRankings).toHaveBeenCalled()
});

// UserGenderTabContent.propTypes = {
//   userId: PropTypes.number,
//   sex: PropTypes.string
// }
