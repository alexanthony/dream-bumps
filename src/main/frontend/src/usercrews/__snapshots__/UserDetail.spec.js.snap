exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="User Details" />
  <CardText>
    <Table
      allRowsSelected={false}
      fixedFooter={true}
      fixedHeader={true}
      height="inherit"
      multiSelectable={false}
      selectable={false}>
      <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={false}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={Object {}}>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Username
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            UserName
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            College
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            New College
          </TableRowColumn>
        </TableRow>
      </TableBody>
    </Table>
  </CardText>
</Card>
`;
